### PR TITLE
This is a desperate attempt to make TensorCache a little more stable

### PR DESCRIPTION
**Pull request by dirkgr**

### CODE REVIEW

File: allennlp/common/file_utils.py [hunk 1]
```diff
@@ -641,8 +641,8 @@ def __init__(
             str(filename),
             subdir=False,
             map_size=map_size,
-            max_readers=cpu_count * 2,
-            max_spare_txns=cpu_count * 2,
+            max_readers=cpu_count * 4,
```
Comment (AkshitaB) [inline allennlp/common/file_utils.py:644]: Should we have an option to specify the kwargs in `TensorCache`'s init? Or do we want to shield the users from the gory details?

**Review by AkshitaB — COMMENTED** [review]
Looks fine. Just a question about how we specify the arguments.

File: allennlp/common/file_utils.py (hunk 1 above)
Comment (dirkgr) [reply allennlp/common/file_utils.py:644]: I wanted to shield users from this. It seems like an implementation detail. Another implementation might not have that setting at all.

Comment (AkshitaB) [reply allennlp/common/file_utils.py:644]: Makes sense.

**Review by AkshitaB — APPROVED** [review]

**Pull request merged**